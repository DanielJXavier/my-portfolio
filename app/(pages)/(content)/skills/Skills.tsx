"use client";

import { useState } from "react";

import Title from "@/_components/Title";
import Psychology from "@/_icons/Psychology";

import Menu from "../_components/Menu";

import shuffle from "./_utils/shuffle";

import { skills as allSkills } from "./data";

const mapSizeToFontSize = {
  1: "text-sm md:text-base lg:text-lg",
  2: "text-base md:text-xl lg:text-2xl",
  3: "text-xl md:text-2px	lg:text-3xl",
  4: "text-2xl md:text-4xl lg:text-5xl",
  5: "text-3xl md:text-5xl lg:text-6xl",
};

export default function Skills() {
  const [skills, setSkills] = useState(allSkills);
  const [filter, setFilter] = useState("all");

  const handleClick = (item: string) => {
    setFilter(item);

    const filteredSkills = allSkills.filter(
      ({ type }) => item === "all" || type === item
    );

    setSkills(filteredSkills);
  };

  return (
    <>
      <Title icon={<Psychology />}>Skills</Title>
      <Menu
        items={["all", "hard", "soft"]}
        activeItem={filter}
        handleClick={handleClick}
      />
      <article className="my-6 md:my-8 lg:my-10 flex gap-x-3 md:gap-x-5 lg:gap-x-8 gap-y-2 md:gap-y-3 lg:gap-y-4 flex-wrap items-center justify-center lg:justify-between">
        {shuffle(skills).map(({ name, size }, i) => (
          <span
            className={`${mapSizeToFontSize[size]} font-light tracking-tighter [word-spacing:-0.35em]`}
            key={i}
            suppressHydrationWarning
          >
            {name}
          </span>
        ))}
      </article>
      <article className="py-5 md:py-6 lg:py-8 text-base md:text-lg lg:text-2xl font-light text-center lg:text-justify">
        And always learning new things to expand the list...
      </article>
    </>
  );
}
