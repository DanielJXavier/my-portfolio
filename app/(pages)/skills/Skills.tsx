"use client";

import { useState } from "react";

import Title from "@/components/Title";
import Psychology from "@/icons/Psychology";

import Menu from "../components/Menu";

import shuffleSkills from "./utils/shuffle";

import { skills } from "./data";

const mapSizeToFontSize = {
  1: "text-sm md:text-base lg:text-lg",
  2: "text-base md:text-xl lg:text-2xl",
  3: "text-xl md:text-2px	lg:text-3xl",
  4: "text-2xl md:text-4xl lg:text-5xl",
  5: "text-3xl md:text-5xl lg:text-6xl",
};

export default function Skills() {
  const [skillFilter, setSkillFilter] = useState("all");

  return (
    <>
      <Title icon={<Psychology />}>Skills</Title>
      <Menu
        items={["all", "hard", "soft"]}
        activeItem={skillFilter}
        setActiveItem={setSkillFilter}
      />
      <article className="my-6 md:my-8 lg:my-10 flex gap-x-3 md:gap-x-5 lg:gap-x-8 gap-y-2 md:gap-y-3 lg:gap-y-4 flex-wrap items-center justify-center lg:justify-between">
        {shuffleSkills(skills)
          .filter(({ type }) => skillFilter === "all" || type === skillFilter)
          .map(({ name, size }, i) => (
            <span
              className={`${mapSizeToFontSize[size]} font-light tracking-tighter [word-spacing:-0.35em]`}
              key={i}
              suppressHydrationWarning
            >
              {name}
            </span>
          ))}
      </article>
      <article className="py-5 md:py-6 lg:py-8 text-base md:text-lg lg:text-2xl font-light text-center lg:text-justify">
        And always learning new things to expand the list...
      </article>
    </>
  );
}
