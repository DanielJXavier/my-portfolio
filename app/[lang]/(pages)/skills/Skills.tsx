"use client";

import { useRef, useState } from "react";

import { useParams, useSearchParams } from "next/navigation";

import { getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import Title from "@/_components/Title";
import Psychology from "@/_icons/Psychology";

import Menu from "../_components/Menu";

import { hardSkills, softSkills as softSkillsKeys } from "./_config";
import shuffle from "./_utils/shuffle";

const mapSizeToFontSize = {
  1: "text-sm md:text-base lg:text-lg",
  2: "text-base md:text-xl lg:text-2xl",
  3: "text-xl md:text-2px	lg:text-3xl",
  4: "text-2xl md:text-4xl lg:text-5xl",
  5: "text-3xl md:text-5xl lg:text-6xl",
};

export default function Skills() {
  const { lang } = useParams<{ lang: Lang }>();
  const searchParams = useSearchParams();

  const random = searchParams.get("random");

  const {
    skills: { title, menu: menuStrings, soft: softSkillsStrings, bottomText },
  } = getDictionary(lang);

  const softSkills = useRef(
    softSkillsKeys.map((skill) => ({
      ...skill,
      name: softSkillsStrings[skill.key],
    }))
  );

  const allSkills = useRef([...hardSkills, ...softSkills.current]);

  const [skills, setSkills] = useState(allSkills.current);
  const [filter, setFilter] = useState("all");

  const menuItems = useRef([
    { text: menuStrings.all, value: "all" },
    { text: menuStrings.hard, value: "hard" },
    { text: menuStrings.soft, value: "soft" },
  ]);

  const handleClick = (item: string) => {
    setFilter(item);

    if (item === "hard") {
      setSkills(hardSkills);
    } else if (item === "soft") {
      setSkills(softSkills.current);
    } else {
      setSkills(allSkills.current);
    }
  };

  return (
    <>
      <Title icon={<Psychology />}>{title}</Title>
      <Menu
        items={menuItems.current}
        activeItem={filter}
        handleClick={handleClick}
      />
      <article className="my-6 md:my-8 lg:my-10 flex gap-x-3 md:gap-x-5 lg:gap-x-8 gap-y-2 md:gap-y-3 lg:gap-y-4 flex-wrap items-center justify-center lg:justify-between">
        {(random === "0" ? skills : shuffle(skills)).map(
          ({ key, name, size }) => (
            <span
              className={`${mapSizeToFontSize[size]} font-light tracking-tighter [word-spacing:-0.35em]`}
              key={key}
              suppressHydrationWarning
            >
              {name}
            </span>
          )
        )}
      </article>
      <article className="py-5 md:py-6 lg:py-8 text-base md:text-lg lg:text-2xl font-light text-center lg:text-justify">
        {bottomText}
      </article>
    </>
  );
}
