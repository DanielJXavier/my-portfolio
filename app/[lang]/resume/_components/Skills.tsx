import { useRef } from "react";

import { useParams } from "next/navigation";

import { getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import {
  hardSkills as allHardSkills,
  softSkills as softSkillsKeys,
} from "@/[lang]/(pages)/skills/_config";

import { chosenHardSkills, chosenSoftSkills } from "../_config";

export default function Skills() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    skills: { soft: softSkillsStrings },
    resume: { sectionTitles },
  } = getDictionary(lang);

  const hardSkills = useRef(
    allHardSkills.filter(({ name }) =>
      chosenHardSkills.some((hardSkill) => name === hardSkill)
    )
  );

  const softSkills = useRef(
    softSkillsKeys
      .filter(({ key }) =>
        chosenSoftSkills.some((softSkill) => key === softSkill)
      )
      .map((skill) => ({
        ...skill,
        name: softSkillsStrings[skill.key],
      }))
  );

  const skills = useRef([...hardSkills.current, ...softSkills.current]);

  return (
    <section>
      <h2 className="text-[16pt] font-bold">{sectionTitles.skills}</h2>
      <ul className="flex gap-x-3.5 flex-wrap">
        {skills.current.map(({ key, name }) => (
          <li
            key={key}
            className="relative text-[12pt] after:content-['â€¢'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
          >
            {name}
          </li>
        ))}
      </ul>
    </section>
  );
}
