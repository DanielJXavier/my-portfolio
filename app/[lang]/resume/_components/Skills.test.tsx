import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import Skills from "./Skills";

import { useParams } from "next/navigation";

jest.mock("next/navigation");

(useParams as jest.Mock).mockImplementation(() => ({ lang: "en" }));

describe("Skills component", () => {
  it("renders the component", () => {
    const { container } = render(<Skills />);

    expect(container).toMatchSnapshot();
  });
});
