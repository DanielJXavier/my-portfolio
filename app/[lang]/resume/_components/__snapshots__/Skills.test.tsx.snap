// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills component renders the component 1`] = `
<div>
  <section>
    <h2
      class="text-[16pt] font-bold"
    >
      Skills
    </h2>
    <ul
      class="flex gap-x-3.5 flex-wrap"
    >
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        HTML5
      </li>
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        CSS3
      </li>
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        JavaScript
      </li>
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        React
      </li>
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        Node.js
      </li>
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        Leadership
      </li>
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        Problem Solving
      </li>
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        Decision Making
      </li>
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        Conflict Resolution
      </li>
      <li
        class="relative text-[12pt] after:content-['•'] after:absolute after:-right-[10px] last-of-type:after:content-['']"
      >
        Resilience
      </li>
    </ul>
  </section>
</div>
`;
